lib/Service/ImportService.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+
+namespace OCA\Tables\Service;
+
+use OC\User\NoUserException;
+use OCA\Tables\Db\Column;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\Service\ColumnTypes\TextLineBusiness;
+use OCP\DB\Exception;
+use OCP\Files\IRootFolder;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
+use OCP\IConfig;
+use OCP\Server;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use PhpOffice\PhpSpreadsheet\Worksheet\Row;
+use PhpOffice\PhpSpreadsheet\Worksheet\Worksheet;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+
+class ImportService extends SuperService {
+
+	private IRootFolder $rootFolder;
+	private ColumnService $columnService;
+	private RowService $rowService;
+
+	private int $tableId = -1;
+	private array $columns = [];
+	private bool $createUnknownColumns = true;
+	private int $countCreatedColumns = 0;
+	private int $countInsertedRows = 0;
+	private int $countErrors = 0;
+
+	public function __construct(PermissionsService $permissionsService, LoggerInterface $logger, ?string $userId,
+		IRootFolder $rootFolder, ColumnService $columnService, RowService $rowService) {
+		parent::__construct($logger, $userId, $permissionsService);
+		$this->rootFolder = $rootFolder;
+		$this->columnService = $columnService;
+		$this->rowService = $rowService;
+	}
+
+	/**
+	 * @throws ContainerExceptionInterface
+	 * @throws NotFoundExceptionInterface
+	 * @throws InternalError
+	 * @throws NotFoundError
+	 */
+	public function import(int $tableId, string $path, bool $createMissingColumns = true): array {
+		if (!$this->userId) {
+			$error = 'No user in context, can not import data. Cancel.';
+			$this->logger->debug($error);
```

Comment:
Sounds more like something to log at a more serious level

Reply:
If the userId is only passed in through dependency injection we can safely assume that the user exists I'd say since the server code would only inject that if a valid user session exists.

Reply:
However if this server might be reused for importing through occ or in a background job it might be good to check against IUserManager::get